airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -284,7 +308,6 @@ public WebBackendConnectionRead webBackendGetConnection(final WebBackendConnecti
     final CatalogDiff diff;
     final AirbyteCatalog syncCatalog;
     if (refreshedCatalog.isPresent()) {
-      connection.setSourceCatalogId(refreshedCatalog.get().getCatalogId());
```

Comment:
This is where I'm removing setting the connection's sourceCatalogId to be the catalogId of the newly discovered catalog. This newly discovered catalog does not yet belong to this connection. Question for @tealjulia and @edmundito though - is this new catalog id necessary for the front end? If it is, we can handle this differently.

Reply:
We don't use the sourceCatalogId, however we want to make sure that it's only associated with the connection after a user clicks "save" on the connection.  A user might refresh but then cancel the changes.  I believe that's why we were passing it to the FE then back again.

Reply:
When a user click's save, do you send this sourceCatalogId back to us so we can update the connection to be the correct sourceCatalogId? I'm testing this out locally, so maybe I can answer my own question pretty soon haha

Reply:
We should be... though now that you ask I'm having trouble finding where.

Reply:
In testing, it does look like we send the new sourceCatalogId to the FE (even though it doesn't officially belong to the connection yet), and then the FE sends it back to us when the user hits save so we can assign that catalog to the connection. It's kind of a weird flow, but I'm not sure it's worth changing now. But it means that I'll have to change how this is done here

Reply:
@tealjulia I've updated this, so this endpoint will continue to pass back the newly refreshed source catalog id on the connection object, even if this source catalog id is not yet saved to the connection.